src/command.rs

```diff
@@ -558,6 +641,11 @@ impl<'l, const C: usize> TryFrom<&'l Data<C>> for Register<'l> {
         info_now!("parsed secret {:?}", &secret);
 
         let kind: oath::Kind = secret_header[0].try_into()?;
+        if kind == Kind::Hmac {
+            // TODO encode verification logic into separate types
+            ensure(secret.len() == 20, FAILED_PARSING_ERROR)?;
+        }
+
         let algorithm: oath::Algorithm = secret_header[0].try_into()?;
```

Comment:
If kind is hmac thes should be checked to be `Sha1` no?

Reply:
Yes, it should be verified on creation, since use is implemented only for SHA1.